File: sdk/trust.py
"""Trust score client for AgentTrust Protocol."""
import httpx


class TrustScoreClient:
    def __init__(self, server_url: str, agent_did: str):
        self.server_url = server_url.rstrip("/")
        self.agent_did = agent_did
        from urllib.parse import quote
        self._encoded_did = quote(agent_did, safe="")

    async def get_score(self) -> dict:
        async with httpx.AsyncClient() as client:
            r = await client.get(f"{self.server_url}/trust/{self._encoded_did}/score")
            r.raise_for_status()
            return r.json()

    async def get_history(self, from_date: str = None, to_date: str = None) -> dict:
        params = {}
        if from_date:
            params["from"] = from_date
        if to_date:
            params["to"] = to_date
        async with httpx.AsyncClient() as client:
            r = await client.get(
                f"{self.server_url}/trust/{self._encoded_did}/history",
                params=params,
            )
            r.raise_for_status()
            return r.json()

    async def recalculate(self) -> dict:
        async with httpx.AsyncClient() as client:
            r = await client.post(
                f"{self.server_url}/trust/{self._encoded_did}/recalculate"
            )
            r.raise_for_status()
            return r.json()

File: sdk-ts/src/trust.ts
export class TrustScoreClient {
  constructor(private serverUrl: string, private agentDid: string) {}

  private get encodedDid(): string {
    return encodeURIComponent(this.agentDid);
  }

  async getScore(): Promise<Record<string, unknown>> {
    const r = await fetch(`${this.serverUrl}/trust/${this.encodedDid}/score`);
    if (!r.ok) throw new Error(await r.text());
    return r.json();
  }

  async getHistory(params?: { from?: string; to?: string }): Promise<Record<string, unknown>> {
    const url = new URL(`${this.serverUrl}/trust/${this.encodedDid}/history`);
    if (params?.from) url.searchParams.set('from', params.from);
    if (params?.to) url.searchParams.set('to', params.to);
    const r = await fetch(url.toString());
    if (!r.ok) throw new Error(await r.text());
    return r.json();
  }

  async recalculate(): Promise<Record<string, unknown>> {
    const r = await fetch(`${this.serverUrl}/trust/${this.encodedDid}/recalculate`, { method: 'POST' });
    if (!r.ok) throw new Error(await r.text());
    return r.json();
  }
}
